pex/dist_metadata.py

```diff
@@ -23,12 +39,40 @@ def requires_python(dist):
     :param dist: A distribution to check for `Python-Requires` metadata.
     :return: The required python version specifiers.
     """
-    if not dist.has_metadata(DistInfoDistribution.PKG_INFO):
+    pkg_info = _parse_pkg_info(dist)
+    if pkg_info is None:
         return None
 
-    metadata = dist.get_metadata(DistInfoDistribution.PKG_INFO)
-    pkg_info = email.parser.Parser().parsestr(metadata)
-    python_requirement = pkg_info.get("Requires-Python")
-    if not python_requirement:
+    python_requirement = pkg_info.get("Requires-Python", None)
+    if python_requirement is None:
         return None
     return SpecifierSet(python_requirement)
+
+
+def requires_dists(
+    dist,  # type: Distribution
+    include_1_1_requires=True,  # type: bool
+):
+    # type: (...) -> Iterator[Requirement]
+    """Examines dist for and returns any declared requirements.
+
+    Looks for `Requires-Dist` metadata and, optionally, the older `Requires` metadata if
+    `include_1_1_requires`.
+
+    See:
+    + https://www.python.org/dev/peps/pep-0345/#requires-dist-multiple-use
+    + https://www.python.org/dev/peps/pep-0314/#requires-multiple-use
+
+    :param dist: A distribution to check for requirement metadata.
+    :return: All requirements found.
+    """
+    pkg_info = _parse_pkg_info(dist)
+    if pkg_info is None:
+        return
```

Comment:
Unsure of the error handling strategy in here, but is a dist not having PKG-INFO something that should be silently ignored like this?

Reply:
I think so, yes. If the distribution has no metadata, that means it expresses no opinon on a Python interpreter requirement, which is all we care about (it can have metadata and not an express an opinion on this field as tests exercise). It is true that distributions should always have metadata and that if they do not, a minimal file should be produced by Pip when it installs the distribution, but there is a range of interesting non-compliant package material out on PyPI and banking on Pip to fill in minimal metadata for those seems like a bet we don't need to make.